project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/DataManagerVault.java

```diff
@@ -0,0 +1,198 @@
+          "Cannot find value for environment variable: %s".formatted(vaultEntryPassword)
+      );
+    }
+    this.envVarKeystoreEntryPassword = vaultEntryPassword;
+    this.envVarKeystorePassword = vaultKeyEnvVar;
+
+    double entropy;
+    if ((entropy = calculateEntropy(System.getenv(envVarKeystorePassword))) < MIN_ENTROPY) {
+      throw new DataManagerVaultException(
+          "Entry of password for keystore was to low: %f (min: %f)".formatted(entropy,
+              MIN_ENTROPY));
+    }
+    if ((entropy = calculateEntropy(System.getenv(envVarKeystoreEntryPassword))) < MIN_ENTROPY) {
+      throw new DataManagerVaultException(
+          "Entry of password for keystore entries was to low: %f (min: %f)".formatted(entropy,
+              MIN_ENTROPY));
+    }
+
+    this.keystorePath = fromString(vaultPathString);
+    this.keyStore = createVault(vaultKeyEnvVar, keystorePath);
+  }
+
+  // Calculates the product of Shannon entropy and secret length
+  // See https://en.wikipedia.org/wiki/Entropy_(information_theory)
+  private static double calculateEntropy(String secret) {
+    if (secret == null || secret.isEmpty()) {
+      return 0.0;
+    }
+
+    Map<Character, Integer> frequencyMap = new HashMap<>();
+    int length = secret.length();
+
+    // Count character frequencies
+    for (char c : secret.toCharArray()) {
+      frequencyMap.put(c, frequencyMap.getOrDefault(c, 0) + 1);
+    }
+
+    // Compute entropy
+    double entropy = 0.0;
+    for (Integer count : frequencyMap.values()) {
+      double probability = (double) count / length;
+      entropy += probability * (Math.log(probability) / Math.log(2));
+    }
+
+    return -entropy * secret.length(); // Negate since log probabilities are negative
+  }
+
+  private static Path fromString(String path) {
+    Path p = Paths.get(path);
+    if (p.isAbsolute()) {
+      return p;
+    }
+    return Path.of(System.getProperty("user.dir")).resolve(path);
+  }
+
+  private static KeyStore createVault(String vaultKeyEnvVar, Path vaultPath)
+      throws CertificateException, KeyStoreException, IOException, NoSuchAlgorithmException {
+
+    return KeyStore.getInstance(vaultPath.toFile(),
+        System.getenv(vaultKeyEnvVar).toCharArray());
+  }
+
+  /**
+   * Adds a secret under a given alias to the vault and stores the vault content into the configured
+   * file.
+   * <p>
+   * {@link DataManagerVault} applies an AES encryption on the provided secret.
+   *
+   * @param alias  the reference the entry can be retrieved again with
+   *               {@link DataManagerVault#read(String)}.
+   * @param secret the secret to store in the vault
+   * @since 1.8.0
+   */
+  public void add(String alias, String secret) {
+    try {
+      this.keyStore.setEntry(alias, new SecretKeyEntry(new SecretKeySpec(secret.getBytes(
+              StandardCharsets.UTF_8), KEY_GENERATOR_ALGORITHM)),
+          new PasswordProtection(System.getenv(envVarKeystoreEntryPassword).toCharArray()));
```

Comment:
why do you use getenv here? As I understand your code the value is injected up top instead of having to get if of the environment all the time?